### PR TITLE
feat(server-mongo-testing): Add Junit 5 replacement for MongoDbRule

### PR DESCRIPTION
**Pull request by RappC**

### CODE REVIEW

File: sda-commons-server-mongo-testing/README.md [hunk 1]
```diff
@@ -2,25 +2,26 @@
...
 ## Usage
 
 To use the test rule, a dependency to this module has to be added:
 
 ```
-testCompile 'org.sdase.commons:sda-commons-server-mongo-testing:<current-version>'
+testCompile 'org.sdase.commons:sda-commons-server-mongo-testing'
 ```
 
+## JUnit 4
```
Comment (Fox32) [inline sda-commons-server-mongo-testing/README.md:24]: Shouldn't this start with `###` as it is a sub-heading of "Usage"? I was wondering why this was changed below

File: sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/MongoDb.java [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+  /**
+   * @return the hostname and port that can be used to connect to the database. The result may
+   *     contain a comma separated list of hosts as in the MongoDB Connection String
+   * @deprecated in favor of {@link #getHosts()}
+   */
+  @Deprecated
+  default String getHost() {
+    return getHosts();
+  }
```
Comment (Fox32) [inline sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/MongoDb.java:52]: As this is deprecated, we shouldn't provide it in `MongoDbClassExtension`

File: sda-commons-server-mongo-testing/README.md (hunk 1 above)
Comment (RappC) [reply sda-commons-server-mongo-testing/README.md]: updated

File: sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/MongoDb.java (hunk 2 above)
Comment (RappC) [reply sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/MongoDb.java]: Absolutely correct. I updated the code accordingly.

File: sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtension.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  @Override
+  public void beforeAll(ExtensionContext context) {
+    logConnection();
+  }
+
+  @Override
+  public void afterAll(ExtensionContext context) {
+    // do nothing
```
Comment (christopher-cudennec) [inline sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtension.java:19]: Maybe we should at least log something?

File: sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      MongoDbClassExtension.builder()
+          .withDatabase("testDb")
+          .withUsername("testuser")
+          .withPassword("testpassword")
+          .build();
+
+  private MongoDbClassExtension useExistingMongoDbClassExtension;
+
+  @BeforeEach
+  public void initUseExistingMongoDbRule() {
```
Comment (christopher-cudennec) [inline sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java:28]: Method still uses "Rule"

File: sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            + EXTERNAL_DB.getUsername()
+            + ":"
+            + EXTERNAL_DB.getPassword()
+            + "@"
+            + EXTERNAL_DB.getHosts()
+            + "/"
+            + EXTERNAL_DB.getDatabase()
+            + "?"
+            + EXTERNAL_DB.getOptions();
+    System.getenv().get(OVERRIDE_MONGODB_CONNECTION_STRING_ENV_NAME);
```
Comment (christopher-cudennec) [inline sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java:41]: I think you wanted to store the old value to be able to restore it after the test, right? :)

File: sda-commons-server-mongo-testing/README.md [hunk 6]
```diff
@@ -54,14 +55,66 @@ private static final DropwizardAppRule<AppConfiguration> DW =
...
 
+
+### JUnit 5
+This module provides the [`MongoDbExtension`](src/main/java/org/sdase/commons/server/mongo/testing/MongoDbExtension.java),
+a JUnit 5 test extension that is used to automatically bootstrap a MongoDB instance for integration tests.
+
+This is accomplished using [Flapdoodle embedded MongoDB](https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo),
+that downloads and starts MongoDB in a separate process.
+As an alternative, one can use [FakeMongo Fongo](https://github.com/fakemongo/fongo) to create an
+in-memory MongoDB database, however it seems to be discontinued and is lacking major features.
```
[the remark this answers is not in the document]
Comment (RappC) [reply sda-commons-server-mongo-testing/README.md:71]: I'll remove it

File: sda-commons-server-mongo-testing/README.md [hunk 7]
```diff
@@ -54,14 +55,66 @@ private static final DropwizardAppRule<AppConfiguration> DW =
...
+
+
+To create a MongoDB instance, add the MongoDB test extension to your test class:
+
+```java
+@RegisterExtension
+static final MongoDbClassExtension MONGO_DB_EXTENSION = MongoDbClassExtension
```
[the remark this answers is not in the document]
Comment (RappC) [reply sda-commons-server-mongo-testing/README.md:78]: done

File: sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDb.java [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class UseExistingMongoDb {
+
+  private static final Logger LOG = LoggerFactory.getLogger(UseExistingMongoDb.class);
+
+  private final String host;
```
[the remark this answers is not in the document]
Comment (RappC) [reply sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDb.java:16]: done

File: sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/StartLocalMongoDbClassExtension.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * JUnit Test extension for running a MongoDB instance alongside the (integration) tests. Can be
+ * configured with custom user credentials and database name. Use {@link #getHosts()} to retrieve
+ * the host to connect to.
+ *
+ * <p>Example usage:
+ *
+ * <pre>
+ * &#64;ClassRule
+ * public static final MongoDbRule RULE = MongoDbRule
```
[the remark this answers is not in the document]
Comment (RappC) [reply sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/StartLocalMongoDbClassExtension.java:15]: Changed comment completely as this class will never be instantiated directly from tests.

File: sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtension.java (hunk 3 above)
Comment (RappC) [reply sda-commons-server-mongo-testing/src/main/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtension.java:19]: removed the method completely

File: sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java (hunk 4 above)
Comment (RappC) [reply sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java:28]: renamed

File: sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java (hunk 5 above)
Comment (RappC) [reply sda-commons-server-mongo-testing/src/test/java/org/sdase/commons/server/mongo/testing/UseExistingMongoDbClassExtensionTest.java:41]: oh yes

File: sda-commons-server-mongo-testing/README.md [hunk 10]
```diff
@@ -54,14 +53,64 @@ private static final DropwizardAppRule<AppConfiguration> DW =
...
+      .withPassword(DATABASE_PASSWORD)
+      .build();
+```
+
+The test extension takes care to choose a free port for the database. You can access the database
+servers address using `MONGO_DB_EXTENSION.getHosts()`.
+Often one need to pass the server address to the constructor of another extension:
+
+```java
+import org.sdase.commons.server.testing.junit5.DropwizardAppExtension;
```
Comment (christopher-cudennec) [inline sda-commons-server-mongo-testing/README.md:87]: The import is no longer needed.

Comment (RappC) [reply sda-commons-server-mongo-testing/README.md:87]: In line 97 the class is used

**Review by christopher-cudennec — APPROVED** [review]
🎸

**Pull request merged**